backend/src/db/repositories/complianceRepository.ts

```diff
@@ -47,6 +51,128 @@ const createAggregationPipeline = (limit: number, offset: number): any[] => {
   return aggregationPipeline;
 };
 
+const softwareDetailAggregationPipeline = (softwareName: string): any[] => [
+  {
+    $match: { "softwareName": softwareName }
+  },
+  {
+    $unwind: "$compliance"
+  },
+  {
+    $project: {
+      softwareName: 1,
+      logo: 1,
+      website: 1,
+      documentation: 1,
+      pointOfContact: 1,
+      version: "$compliance.version",
+      bbDetails: {
+        $arrayToObject: {
+          $map: {
+            input: { $objectToArray: "$compliance.bbDetails" },
+            as: "bbDetail",
+            in: [
+              "$$bbDetail.k",
+              {
+                bbVersion: "$$bbDetail.v.bbVersion",
+                requirementSpecificationCompliance: {
+                  level: "$$bbDetail.v.requirementSpecificationCompliance.level",
+                  note: "$$bbDetail.v.requirementSpecificationCompliance.note"
+                },
+                interfaceCompliance: {
+                  level: "$$bbDetail.v.interfaceCompliance.level",
+                  note: "$$bbDetail.v.interfaceCompliance.note"
+                }
+              }
+            ]
+          }
+        }
+      }
+    }
+  },
+  {
+    $group: {
+      _id: "$softwareName",
+      logo: { $first: "$logo" },
+      website: { $first: "$website" },
+      documentation: { $first: "$documentation" },
+      pointOfContact: { $first: "$pointOfContact" },
```

Comment:
Just a note that this may not be ideal solution for the flow and we should decide on how data should be displayed when more that one form was submitted for given software.